main.rb

```diff
@@ -2,36 +2,36 @@ class Main
   def start_console
     puts 'Welcome to Catalog App'
 
-    loop_method
-  end
-
-  def select_an_option
-    {
-      'List all books' => 'list_books',
-      'List all music albums' => 'list_music_albums',
-      'List all movies' => 'list_movies',
-      'List all games' => 'list_games',
-      'List all genres' => 'list_genres',
-      'List all labels' => 'list_labels',
-      'List all authors' => 'list_authors',
-      'List all sources' => 'list_sources',
-      'Add a book' => 'add_book',
-      'Add a music album' => 'add_music_album',
-      'Add a movie' => 'add_movie',
-      'Add a game' => 'add_game'
+    options = {
+      'List all books' => :list_books,
+      'List all music albums' => :list_music_albums,
+      'List all movies' => :list_movies,
+      'List all games' => :list_games,
+      'List all genres' => :list_genres,
+      'List all labels' => :list_labels,
+      'List all authors' => :list_authors,
+      'List all sources' => :list_sources,
+      'Add a book' => :add_book,
+      'Add a music album' => :add_music_album,
+      'Add a movie' => :add_movie,
+      'Add a game' => :add_game
     }
+
+    loop_method(options)
   end
 
-  def loop_method
-    options = select_an_option
+  def loop_method(options)
+    puts 'Please choose an option by selecting a number'
     loop do
-      options.each_with_index do |option, index|
-        puts "option #{index + 1} : #{option[0]}"
+      options.each_with_index do |(option, _), index|
+        puts "#{index + 1}) #{option}"
       end
 
       puts 'Insert any key to exit'
-      input = gets.chomp
-      if input >= 13
+      input = gets.chomp.to_i
+      if input.positive? && input <= 13
+        options.keys[input]
```

Comment:
Hello, @NahnahAJ.

You made some good changes. The interface is now more appealing than before. There is, however, one minor issue.

By using `options.keys[input]` you are searching for hash keys rather than values. Instead, use `option.values[input]` to call the correct function.

Also, remember hashes began with 0 rather than 1, you must subtract 1 from the user input.

Reply:
Hey @arabiu033 when I subtract -1 from the user input, It throws the app into a minor bug. So I have decided to pause on that. We can change it when we are done with the program. Or tomorrow God willing. But I have implemented the other change you requested.